graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -426,37 +426,39 @@ auto GetPickFn(
   };
 }
 
+template <SamplerType S>
 auto GetTemporalPickFn(
     torch::Tensor seed_timestamp, torch::Tensor csc_indices,
     const std::vector<int64_t>& fanouts, bool replace,
     const torch::TensorOptions& options,
     const torch::optional<torch::Tensor>& type_per_edge,
     const torch::optional<torch::Tensor>& probs_or_mask,
     const torch::optional<torch::Tensor>& node_timestamp,
-    const torch::optional<torch::Tensor>& edge_timestamp) {
-  return [&seed_timestamp, &csc_indices, &fanouts, replace, &options,
-          &type_per_edge, &probs_or_mask, &node_timestamp, &edge_timestamp](
-             int64_t seed_offset, int64_t offset, int64_t num_neighbors,
-             auto picked_data_ptr) {
-    // If fanouts.size() > 1, perform sampling for each edge type of each
-    // node; otherwise just sample once for each node with no regard of edge
-    // types.
-    if (fanouts.size() > 1) {
-      return TemporalPickByEtype(
-          seed_timestamp, csc_indices, seed_offset, offset, num_neighbors,
-          fanouts, replace, options, type_per_edge.value(), probs_or_mask,
-          node_timestamp, edge_timestamp, picked_data_ptr);
-    } else {
-      int64_t num_sampled = TemporalPick(
-          seed_timestamp, csc_indices, seed_offset, offset, num_neighbors,
-          fanouts[0], replace, options, probs_or_mask, node_timestamp,
-          edge_timestamp, picked_data_ptr);
-      if (type_per_edge) {
-        std::sort(picked_data_ptr, picked_data_ptr + num_sampled);
-      }
-      return num_sampled;
-    }
-  };
+    const torch::optional<torch::Tensor>& edge_timestamp, SamplerArgs<S> args) {
+  return
+      [&seed_timestamp, &csc_indices, &fanouts, replace, &options,
+       &type_per_edge, &probs_or_mask, &node_timestamp, &edge_timestamp, args](
+          int64_t seed_offset, int64_t offset, int64_t num_neighbors,
+          auto picked_data_ptr) {
+        // If fanouts.size() > 1, perform sampling for each edge type of each
+        // node; otherwise just sample once for each node with no regard of edge
+        // types.
+        if (fanouts.size() > 1) {
+          return TemporalPickByEtype(
+              seed_timestamp, csc_indices, seed_offset, offset, num_neighbors,
+              fanouts, replace, options, type_per_edge.value(), probs_or_mask,
+              node_timestamp, edge_timestamp, args, picked_data_ptr);
+        } else {
+          int64_t num_sampled = TemporalPick(
+              seed_timestamp, csc_indices, seed_offset, offset, num_neighbors,
+              fanouts[0], replace, options, probs_or_mask, node_timestamp,
+              edge_timestamp, args, picked_data_ptr);
+          if (type_per_edge) {
```

Comment:
If it is a tensor, but all value are 0, will it return false?

Reply:
I think it is a torch::optional, so it is checking if is has a value. I didn't author this code, it was like this before the change.

I can change it to
`if (type_per_edge.value())` if you will like that better.

Reply:
Changed it as suggested above to make it more readable.